Python/compile.c

```diff
@@ -3259,12 +3259,14 @@ static int
 compiler_break(struct compiler *c)
 {
     struct fblockinfo *loop = NULL;
+    int origin_lineno = c->u->u_lineno;
     /* Emit instruction with line number */
     ADDOP(c, NOP);
     if (!compiler_unwind_fblock_stack(c, 0, &loop)) {
         return 0;
     }
     if (loop == NULL) {
+        c->u->u_lineno = origin_lineno;
```

Comment:
compiler_error uses all four elements of the location, so it might be inconsistent if you only update one of them: https://github.com/python/cpython/blob/83b0735e2d61f2cf3d0eeb9c905e6d853cb5223b/Python/compile.c#L6150

Reply:
Ah, I understand what you intended at https://github.com/python/cpython/pull/101447#discussion_r1091731716